### ISSUE TITLE
Ctrl + Shift + U does strange thing, am I stupid?

### ISSUE THREAD

**Issue by MindTooth** [kovidgoyal/kitty#350]
Just tested the new feature, but it works in strange ways.

Am I to expect that when I open the unicode page, I am able to enter a sequence of either one or two character, then press enter, and that selection will be enter into the shell?

A lot of the options does not even work.  Also, adding emojis to the shell are not deletable.

What am I doing wrong?  Tried setting a different font, but still the same.

My system:
<img width="682" alt="screen shot 2018-02-27 at 11 43 29" src="https://user-images.githubusercontent.com/33870508/36724439-7e8fa520-1bb3-11e8-8922-3ba1e7d9023f.png">

Here I've selected a combo that wont expand into anything:
<img width="1920" alt="screen shot 2018-02-27 at 11 43 44" src="https://user-images.githubusercontent.com/33870508/36724440-7eac9126-1bb3-11e8-9f71-4621c24c13cc.png">

**Comment by kovidgoyal** [comment]
In Code mode, you type in either the hex code of the character (for example 2716 for ✖ ) , or if you want to select a character from the list, type the period and then the two letter code, you appear to have forgotten the period, in your screen shot.

**Comment by kovidgoyal** [comment]
And I dont know what you mean by the emojis are not deletable. The emojis are just characters like any other, you delete them by pressing backspace.

**Comment by MindTooth** [comment]
Seems to do the trick.  Thanks.

The delete bug seems to be related the fish-shell I believe.

Getting this error once and while too:
```
Traceback (most recent call last):
  File "kitty/loop.py", line 327, in loop
  File "kitty/main.py", line 333, in initialize
  File "kitty/main.py", line 381, in draw_screen
  File "kitty/main.py", line 216, in layout
ZeroDivisionError: integer division or modulo by zero
```

**Comment by kovidgoyal** [comment]
I doubt many shells have good support for unicode wide characters like emoji, try using the unicode input inside an editor instead, like vim or neovim. That said, emoji input works fine inside zsh. Basically, the shell has to know that the unicode character is two cells wide, which means it needs an implementation of wcwidth() that is based on a recent unicode standard. 

As for that error, this should fix it. https://github.com/kovidgoyal/kitty/commit/e8f3f0079250291d8aea36c63d44fc30bca4ca06

**Comment by MindTooth** [comment]
Tried building the app from source:
```shell
~/S/kitty (master|✔) $ make app
logo/make.py
Rendering at 256x256...
Traceback (most recent call last):
  File "logo/make.py", line 19, in <module>
    render('kitty.png')
  File "logo/make.py", line 15, in render
    subprocess.check_call(['rsvg-convert', '-w', str(sz), '-h', str(sz), '-o', output, src])
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 535, in check_call
    retcode = call(*popenargs, **kwargs)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 522, in call
    return Popen(*popenargs, **kwargs).wait()
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 710, in __init__
    errread, errwrite)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 1335, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory
make: *** [logo/kitty.iconset/icon_256x256.png] Error 1
```

Also, seems like there were a patch to fish-shell [two days](https://github.com/fish-shell/fish-shell/commit/5282d3e7110f40fc9cb51f4ae952f65000bbf0ae) ago that trying to fix the issue.

**Comment by kovidgoyal** [comment]
You're missing rsvg-convert, see the readme under building kitty.app

**Comment by kovidgoyal** [comment]
Oh and since you found the unicode input tool confusing initially, it would be nice if you could contribute a couple of sentences of documentation to the readme to clear it up. Might help others.

**Comment by MindTooth** [comment]
Did not print the error message with the new build.  However, experienced a new error.

```shell
$ make app
python3 setup.py app 
usage: setup.py [-h] [--debug] [-v] [--sanitize] [--prefix PREFIX] [--full]
                [--profile]
                [{build,test,linux-package,kitty.app,osx-bundle,clean}]
setup.py: error: argument action: invalid choice: 'app' (choose from 'build', 'test', 'linux-package', 'kitty.app', 'osx-bundle', 'clean')
make: *** [app] Error 2
```

Installed both dependencies stated in the README.

**Comment by kovidgoyal** [comment]
https://github.com/kovidgoyal/kitty/commit/ee1287211dba7fb6a0cc6b044853a012cd97bb84

**Comment by MindTooth** [comment]
Noticed.  So I pulled the commit, build and replaced the kitty.app inside /Applications.  Now I got:
```shell
Failed to launch child: kitty
With error: Permission denied
Press Enter to exit.
```

Not sure what causes it.  When running kitty from `make && python3 .` I did not have this error.  And I can spawn the the unicode screen several times in a row.  Reverting back to last stable version, and waiting for the changes in the next release.

Thank for the help.
